homeassistant/components/template/light.py

```diff
@@ -451,12 +451,17 @@ def _update_temperature(self, render):
     @callback
     def _update_color(self, render):
         """Update the hs_color from the template."""
-        if render in ("None", ""):
-            self._color = None
-            return
-        h_str, s_str = map(
-            float, render.replace("(", "").replace(")", "").split(",", 1)
-        )
+        h_str = s_str = None
+        if isinstance(render, str):
+            if render in ("None", ""):
+                self._color = None
+                return
+            h_str, s_str = map(
+                float, render.replace("(", "").replace(")", "").split(",", 1)
+            )
+        elif isinstance(render, (set, list, tuple)) and len(render) == 2:
```

Comment:
A set is unordered and so can't be used for unpacking.
```suggestion
        elif isinstance(render, (list, tuple)) and len(render) == 2:
```